pkg/bastion/acl.go

```diff
@@ -46,5 +50,51 @@ func checkACLs(user dbmodels.User, host dbmodels.Host) string {
 	}
 	sort.Sort(byWeight(acls))
 
-	return acls[0].Action
+	return checkACLsHook(aclCheckCmd, acls[0].Action, user, host)
+}
+
+// checkACLsHook executes external command to check ACL and passes following parameters:
+// $1 - SSH Portal action (`allow` or `deny`)
+// $2 - User as JSON string
+// $3 - Host as JSON string
+// External program has to return `allow` or `deny` in stdout.
+func checkACLsHook(aclCheckCmd string, action string, user dbmodels.User, host dbmodels.Host) string {
+	if aclCheckCmd != "" {
+		jsonUser, err := json.Marshal(user)
+		if err != nil {
+			log.Printf("Error: %v", err)
+			return action
+		}
+
+		jsonHost, err := json.Marshal(host)
+		if err != nil {
+			log.Printf("Error: %v", err)
+			return action
+		}
+
+		args := []string{
+			action,
+			string(jsonUser),
+			string(jsonHost),
+		}
+
+		out, err := exec.Command(aclCheckCmd, args...).CombinedOutput()
```

Comment:
This should have a timeout, can you add the timeout or at least a comment for now?

Reply:
added timeout